### DISCUSSION TITLE
Populate portal wifi with saved credentials

### DISCUSSION THREAD

**Discussion by vanceb** [khoih-prog/ESP_WiFiManager#91] [Q&A]
### How can I populate the config portal ssid and password fields from saved credentials?

It is possible to set the initial values for custom parameters by providing the values when you setup the parameter.  How can I get a similar effect for the ssid, ssid1 and password, password1 fields?  They are always blank, even when there are saved credentials!

### Steps to Reproduce

Start the config portal and set the SSID and passwords for 2 different networks.
Make sure that it connects to at least one of the configured networks and that credentials are stored.
Reboot and initiate the config portal again - The saved credentials are not populated into the ssid1, ssid2, password1, password2 fields - they are empty

### Expected behavior

I expect the wifi fields ssid1, ssid2, password1, password2 fields to be populated with the previously saved values

### Actual behavior

The ssid1, ssid2, password1, password2 fields are blank

**Comment by khoih-prog** [comment]
You really don't need to do anything for SSID/PWD fields. If leaving them `blank`, the stored SSIDs/PWDs will be auto reuse. The real reason for the design is that only displaying `Credentials` on CP when necessary to avoid `Security` risk (somebody sees your secrets !!!)

Try and see.

Anyway, check the new [**ESP_WiFiManager** v1.12.0](https://github.com/khoih-prog/ESP_WiFiManager/releases/tag/v1.12.0), with the new **optional** feature

```cpp
// To not display stored SSIDs and PWDs on Config Portal, select false. Default is true
// Even the stored Credentials are not display, just leave them all blank to reconnect and reuse the stored Credentials 
//#define DISPLAY_STORED_CREDENTIALS_IN_CP        false
```
Also a hint is added over the SSID/PWD fields

```
*Hint: To reuse the saved WiFi credentials, leave SSID and PWD fields empty
```

![Configuration](https://user-images.githubusercontent.com/57012152/194677287-8e9a39ef-0e1c-4ec1-a50f-e1dbcbc867e2.png)

---

### Releases v1.12.0

1. Optionally display Credentials (SSIDs, PWDs) in Config Portal. Check [Populate portal wifi with saved credentials #91](https://github.com/khoih-prog/ESP_WiFiManager/discussions/91)
2. Display `Credentials` Hint on Config Portal
3. Periodic code clean-up
